README.md

```diff
@@ -4,7 +4,7 @@ This is a repository of provably-correct versions of Leftpad.
 
 ## What is "provably-correct"?
 
-**Provably correct code** is code that you can totally guarantee does what you say it does. You do this by providing a **proof** that a computer can check. If the proof is wrong, the code won't compile.
+**Provably correct code** is code that you can totally guarantee does what you say it does. You do this by providing a [**proof** that a computer can check](https://en.wikipedia.org/wiki/Formal_methods). If the proof is wrong, the code won't compile.
```

Comment:
Possibly a better link could be [Automated theorem proving](https://en.wikipedia.org/wiki/Automated_theorem_proving), following the discussion in #38 (which, by the way, originally motivated this PR).